topology/probes/istio/virtualservice.go

```diff
@@ -88,20 +89,20 @@ func virtualServicePodMetadata(a, b interface{}, typeA, typeB, manager string) g
 	if pod.Labels["app"] == "" {
 		return m
 	}
-	weight := 0
 	for _, http := range vsSpec.HTTP {
+		if http.Match != nil {
+			continue
+		}
 		for _, route := range http.Routes {
 			if route.Destination.App != pod.Labels["app"] {
 				continue
 			}
 			if route.Destination.Version == pod.Labels["version"] || route.Destination.Version == "" {
-				weight += route.Weight
+				m["weight"] = route.Weight
```

Comment:
I found the below paragraph here https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/:

```
REQUIRED. The proportion of traffic to be forwarded to the service version. (0-100). Sum of weights across destinations SHOULD BE == 100. If there is only one destination in a rule, the weight value is assumed to be 100.
```

Indicating you should set `m["weight"]` to `100` in the above case

Reply:
DONE